File: Source/AlphaTab/Rendering/Effects/PalmMuteEffectInfo.cs
using AlphaTab.Model;
using AlphaTab.Rendering.Glyphs;

namespace AlphaTab.Rendering.Effects
{
    internal class PalmMuteEffectInfo : NoteEffectInfoBase
    {
        public override string EffectId => "palm-mute";

        protected override bool ShouldCreateGlyphForNote(Note note)
        {
            return note.IsPalmMute;
        }

        public override EffectBarGlyphSizing SizingMode => EffectBarGlyphSizing.GroupedOnBeat;

        public override EffectGlyph CreateNewGlyph(BarRendererBase renderer, Beat beat)
        {
            return new LineRangedGlyph("P.M.");
        }
    }
}

File: packages/alphatab/src/rendering/effects/PalmMuteEffectInfo.ts
import type { Beat } from '@coderline/alphatab/model/Beat';
import type { Note } from '@coderline/alphatab/model/Note';
import type { BarRendererBase } from '@coderline/alphatab/rendering/BarRendererBase';
import { EffectBarGlyphSizing } from '@coderline/alphatab/rendering/EffectBarGlyphSizing';
import { NoteEffectInfoBase } from '@coderline/alphatab/rendering/effects/NoteEffectInfoBase';
import type { EffectGlyph } from '@coderline/alphatab/rendering/glyphs/EffectGlyph';
import { LineRangedGlyph } from '@coderline/alphatab/rendering/glyphs/LineRangedGlyph';
import { NotationElement } from '@coderline/alphatab/NotationSettings';

/**
 * @internal
 */
export class PalmMuteEffectInfo extends NoteEffectInfoBase {
    public get notationElement(): NotationElement {
        return NotationElement.EffectPalmMute;
    }

    protected shouldCreateGlyphForNote(note: Note): boolean {
        return note.isPalmMute;
    }

    public get sizingMode(): EffectBarGlyphSizing {
        return EffectBarGlyphSizing.GroupedOnBeat;
    }

    public createNewGlyph(_renderer: BarRendererBase, _beat: Beat): EffectGlyph {
        return new LineRangedGlyph('P.M.', NotationElement.EffectPalmMute);
    }
}
